template/gem_name.gemspec.erb

```diff
@@ -32,5 +32,5 @@ Gem::Specification.new do |spec|
 
   spec.required_ruby_version = '>= 3.0', '< 4'
 
-  spec.add_dependency 'faraday', '~> 2.9'
+  spec.add_dependency 'faraday', '~> 2.9.0'
```

Comment:
Perhaps we wish this to be "accept anything <3.0"?

Reply:
Then we need an additional matrix of Faraday versions in CI.

Reply:
That's a reasonable point, but I'm pretty sure it's not an issue. Bundler checks Ruby requirements in gems and their dependencies, so if someone running Ruby 2.7 tries to install this, it won't pull Faraday 2.10 which only supports Ruby 3.0+

The point on the CI is valid, that's something we can decide on how to deal with. As you suggest, one solution is to extend the matrix to test Ruby 2.7 with Faraday 2.9, but since 2.7 is EOL, my inclination would be to simply drop it and make the template support Ruby 3.0+ as well

Reply:
> That's a reasonable point, but I'm pretty sure it's not an issue. Bundler checks Ruby requirements in gems and their dependencies, so if someone running Ruby 2.7 tries to install this, it won't pull Faraday 2.10 which only supports Ruby 3.0+

Fair enough.

But this is a template.

Imagine you've created a Faraday plugin.

You have, for example, Faraday >= 2.9 and Ruby >= 3.0.

Then you add… no, there should not be fails with dependency bot.

OK, then I see only this issue for now: if Faraday 2.20 drops Ruby 3.0 support, your lib still supports it, depending on Faraday < 2.20, it's OK and should work, but then Faraday 2.30 changes its API, related to your lib, and you change it too, but your lib starts fail on Ruby 3.0 with Faraday < 2.20.

It's pretty hypothetical and not-too-real example, and I agree to change it. But… we should be careful here, and it can bring problems in the future, for the template users.

---

I'll think about Faraday matrix later.

Reply:
Changed to `'>= 2.9', '< 3'`. Due to your opinion that we should keep it with Ruby >= 3.0.